src/contracts/MultiLP.sol

```diff
@@ -0,0 +1,250 @@
+    /// nextWithdrawalIndex - index of the next withdrawal request starting at 0
+    // slither-disable-next-line uninitialized-state
+    WithdrawalQueueMetadata public withdrawalQueueMetadata;
+
+    struct WithdrawalRequest {
+        address withdrawer;
+        bool claimed;
+        // When the withdrawal can be claimed
+        uint40 claimTimestamp;
+        // Amount of assets to withdraw
+        uint128 assets;
+        // cumulative total of all withdrawal requests including this one.
+        // this request can be claimed when this queued amount is less than or equal to the queue's claimable amount.
+        uint128 queued;
+    }
+
+    /// @notice Mapping of withdrawal request indices to the user withdrawal request data
+    mapping(uint256 requestId => WithdrawalRequest) public withdrawalRequests;
+
+    uint256[50] private _gap;
+
+    event RedeemRequested(
+        address indexed withdrawer, uint256 indexed requestId, uint256 assets, uint256 queued, uint256 claimTimestamp
+    );
+    event RedeemClaimed(address indexed withdrawer, uint256 indexed requestId, uint256 assets);
+
+    constructor(address _liquidityAsset) {
+        require(_liquidityAsset == address(token0) || _liquidityAsset == address(token1), "invalid liquidity asset");
+        liquidityAsset = _liquidityAsset;
+    }
+
+    /// @dev called by the concrete contract's `initialize` function
+    function _initMultiLP(string calldata _name, string calldata _symbol) internal {
+        __ERC20_init(_name, _symbol);
+
+        // Transfer a small bit of liquidity from the intializer to this contract
+        IERC20(liquidityAsset).transferFrom(msg.sender, address(this), MIN_TOTAL_SUPPLY);
+
+        // mint a small amount of shares to a dead account so the total supply can never be zero
+        // This avoids donation attacks when there are no assets in the ARM contract
+        _mint(DEAD_ACCOUNT, MIN_TOTAL_SUPPLY);
+    }
+
+    /// @notice Preview the amount of shares that would be minted for a given amount of assets
+    /// @param assets The amount of liquidity assets to deposit
+    /// @return shares The amount of shares that would be minted
+    function previewDeposit(uint256 assets) public view returns (uint256 shares) {
+        shares = convertToShares(assets);
+    }
+
+    /// @notice deposit liquidity assets in exchange for liquidity provider (LP) shares.
+    /// The caller needs to have approved the contract to transfer the assets.
+    /// @param assets The amount of liquidity assets to deposit
+    /// @return shares The amount of shares that were minted
+    function deposit(uint256 assets) external returns (uint256 shares) {
+        _preDepositHook();
+
+        shares = convertToShares(assets);
+
+        // Transfer the liquidity asset from the sender to this contract
+        IERC20(liquidityAsset).transferFrom(msg.sender, address(this), assets);
+
+        // mint shares
+        _mint(msg.sender, shares);
```

Comment:
Typically in AMM's this mint to dead happens during the first deposit. I'm wondering if there are bad effects of doing it where there are zero assets.

Reply:
The problem with doing on the first deposit is the `deposit` function has to have special logic that is only used one. That's why I prefer to put that logic in the initialize function.